src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +250,146 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String},
+     * or a default value.  The extension is defined to be the portion of
+     * the {@code String} representation of the file name after the last
+     * dot ('.').  If the first character in the file name string is a dot
+     * it is ignored.  If the extension cannot be determined, then the
+     * parameter {@code defaultExtension} is returned.  This will occur if
+     * the path has zero elements ({@link #getFileName()} returns
+     * {@code null}), or the file name string does not contain a dot, only
+     * the first character is a dot, or the last character is a dot.
```

Comment:
Musing on a filename with a final dot, i.e.  `FooBar.`
I think it has an empty string "" as the extension (because there is a dot).

As described, if the `defaultExtension` is returned, then it would be nice for the code to trust
that there is no final dot.  If it is case when the `defaultExtension` is returned, there may or may not be a final dot then the caller may need to test separately for it before using the filename.

Reply:
Currently a filename with a terminating dot is considered not to have an extension. Yes, `defaultExtension` should have been checked for a dot.